apps/ejabberd/src/mod_muc_light_commands.erl

```diff
@@ -169,6 +184,24 @@ send_message(Domain, RoomName, Sender, Message) ->
             ejabberd_router:route(S, R, Stanza)
     end.
 
+-spec delete_room(DomainName :: binary(), RoomName :: binary(),
+                  Sender :: binary()) ->
+                         ok | {error, not_exists} | {error, not_allowed}.
+delete_room(DomainName, RoomName, Owner) ->
+    OwnerJID = jid:binary_to_bare(Owner),
+    case muc_light_room_name_to_jid(OwnerJID, RoomName,
+                                    DomainName) of
+        {error, _} = Error ->
+            Error;
+        RoomJID ->
+            case mod_muc_light:is_room_owner(true, RoomJID, OwnerJID) of
```

Comment:
This involves additional call to DB, that can be avoided with improved `find_room_with_name`, as it could return user's affiliation in the room. Besides, it uses hook handler without actually calling a hook.